server/datastore/mysql/migrations/tables/20250410104321_UpdateMacOSSoftwareNames.go

```diff
@@ -76,38 +85,49 @@ WHERE
 		WHERE
 			hs2.software_id = ?
 			AND hs2.host_id = hs1.host_id)`
+	updateHostSoftwareInstalledPathsStmt := `UPDATE host_software_installed_paths SET software_id = ? WHERE software_id IN (?)`
```

Comment:
Above this, `getRecordToUpdateStmt` finds all host IDs in `host_software` that have rows with software IDs that we want to merge, but _doesn't_ already have a row with the software ID we want to merge _to_.  We'll use this to insert a row for the "merged" software ID while avoiding duplicate rows if that rows already exists 👍 

`updateHostSoftwareInstalledPathsStmt` updates the duplicate software IDs to the selected "merged" one.  @iansltx I don't know the implications of duplicate `host_id`/`software_id` keys in this table, possibly with different paths.  Is this what we'd expect if a host had two copies of a software item in different locations (e.g. in their Downloads folder as well as in their Applications folder?)

Reply:
@sgress454 You understand correctly. You can see this in test setup in L109-112, and tested in L205-212.

Normally you wouldn't see this because you wouldn't have the same version of the software in different directories, which is what this would take (different versions would resolve to different software entries), but we want to cover this case properly since we should have the same number of associated paths linked to software that's still in the DB (since we have no fkeys back to software for perf reasons).